File: jira_client.py
import requests
from config import JIRA_BASE_URL, JIRA_EMAIL, JIRA_API_TOKEN

def get_recent_issues(project_keys, days=7, max_results=50):
    url = f"{JIRA_BASE_URL}/rest/api/3/search"
    headers = {
        "Accept": "application/json"
    }
    auth = (JIRA_EMAIL, JIRA_API_TOKEN)

    project_list = ", ".join(project_keys)
    jql = f'project IN ({project_list}) AND updated >= -{days}d ORDER BY updated DESC'

    params = {
        "jql": jql,
        "maxResults": max_results,
        "fields": "summary,status,assignee,updated,project"
    }

    response = requests.get(url, headers=headers, auth=auth, params=params)
    response.raise_for_status()
    return response.json().get("issues", [])

File: src/lib/services/jiraClient.ts
import fetch from 'node-fetch';

// Define Jira issue interface
export interface JiraIssue {
  fields: {
    project: {
      key: string;
    };
    summary: string;
    status: {
      name: string;
    };
    assignee?: {
      displayName: string;
    };
  };
}

export async function getRecentIssues(projects: string[], days: number) {
  const jql = `project IN (${projects.join(',')}) AND updated >= -${days}d ORDER BY updated DESC`;

  const url = new URL(`${process.env.JIRA_BASE_URL}/rest/api/3/search/jql`);
  url.searchParams.append('jql', jql);
  url.searchParams.append('maxResults', '50');
  url.searchParams.append('expand', 'changelog');
  url.searchParams.append('fields', 'summary,status,assignee,updated,project');

  const auth = Buffer.from(`${process.env.JIRA_EMAIL}:${process.env.JIRA_API_TOKEN}`).toString('base64');

  const response = await fetch(url.toString(), {
    headers: {
      'Authorization': `Basic ${auth}`,
      'Accept': 'application/json'
    }
  });

  if (!response.ok) {
    const errorBody = await response.text();
    throw new Error(`JIRA API request failed (${response.status}): ${errorBody}`);
  }

  const data = await response.json();
  return data.issues || [];
}